### PR TITLE
Validate optional formats

### PR DESCRIPTION
**Pull request by robbielamb-wf**
This work allows for formats in drafts 2019-09 and 2020-12 to be validated.

### CODE REVIEW

File: library/src/main/java/net/jimblackler/jsonschemafriend/FormatChecker.java [hunk 1]
```diff
@@ -48,13 +49,13 @@
...
   static String formatCheck(
-      String string, String format, URI metaSchema, RegExPatternSupplier regExPatternSupplier) {
+      String string, String format, URI metaSchema, RegExPatternSupplier regExPatternSupplier, boolean validateFormats) {
     boolean preDraft4 = metaSchema.equals(DRAFT_3);
     boolean preDraft6 = preDraft4 || metaSchema.equals(DRAFT_4);
     boolean preDraft7 = preDraft6 || metaSchema.equals(DRAFT_6);
     boolean preDraft2019 = preDraft7 || metaSchema.equals(DRAFT_7);
 
-    if (!preDraft7 && preDraft2019) {
+    if ((!preDraft7 && preDraft2019) || validateFormats) {
```
Comment (tomdeering-wf) [inline library/src/main/java/net/jimblackler/jsonschemafriend/FormatChecker.java:58]: Do we care about the combination of `preDraft7 == true` and `validateFormats == true`?

**Comment by jimblackler** [comment]
Could you take a look at the presubmit check failures?

```
SuiteTest > draft2019_09() > format > time.json > validation of time strings > net.jimblackler.jsonschemafriend.SuiteTest.draft2019_09()[3][15][1][32] FAILED
    org.opentest4j.AssertionFailedError at SuiteTest.java:152

SuiteTest > draft2020_12() > format > time.json > validation of time strings > net.jimblackler.jsonschemafriend.SuiteTest.draft2020_12()[3][15][1][32] FAILED
    org.opentest4j.AssertionFailedError at SuiteTest.java:152

SuiteTest > draft7() > format > time.json > validation of time strings > net.jimblackler.jsonschemafriend.SuiteTest.draft7()[3][14][1][32] FAILED
    org.opentest4j.AssertionFailedError at SuiteTest.java:152
```

**Comment by robbielamb-wf** [comment]
Looks like these tests are failing because `DateTimeFormatter` changed after java 8. I verified the tests fail on my machine with openjdk-1.8 and are passing on openjdk-1.11 and openjdk-1.17. 

Looks like the options are to:
1. Add the tests to the skip lists, so they are ignored when they fail with java 8.
2. Find an alternative strategy to check the time formats.
3. Change CI to run on a different java version.

Do you have an opinion on how to fix this?

**Comment by jimblackler** [comment]
I'd greatly prefer '2', perhaps a different date/time library if one exists that can make the right checks.

**Comment by robbielamb-wf** [comment]
I've added a regular expression that can check the time format. 
I created another branch that uses the Joda time library to check the format. It works and I can swap that in if you'd prefer to use the library over a regexp.

**Comment by jimblackler** [comment]
If the regex does the job then that's fine.

**Pull request merged**